tests/conftest.py

```diff
@@ -1,11 +1,80 @@
 # built-in
 import shutil
+import socket
 from pathlib import Path
 
 # external
 import pytest
 
 
+true_socket = socket.socket
+true_connect = socket.socket.connect
+
+
+class SocketBlockedError(RuntimeError):
+    pass
+
+
+class SocketConnectBlockedError(RuntimeError):
+    pass
+
+
+def fake_socket(*args, **kwargs):
+    raise SocketBlockedError('use @pytest.mark.allow_hosts to unblock some hosts')
+
+
+def pytest_addoption(parser):
+    group = parser.getgroup('socket')
+    group.addoption(
+        '--no-network',
+        action='store_true',
+        help='skip tests with network interactions',
+    )
+
+
+@pytest.fixture(autouse=True)
+def no_network(request):
```

Comment:
you might want to use pytest-socket , but note that it (and your implementation) are not very effective as noted at https://github.com/miketheman/pytest-socket/issues/25 - `getaddrinfo` is usually called before socket.connect -- and getaddrinfo is where the failure happens in an offline environment.

Reply:
Yes, my implementation based on `pytest-socket` with some ad-hoc improvements. However, it will work: when you're doing tests offline, you're using `--no-network` flag and these tests are skipped, not mocked.

Reply:
https://github.com/dephell/dephell/pull/149/files#diff-484462fced51d1a06b1d93b4a44dd535R57

Reply:
```python
    if request.config.getoption('--no-network'):
        pytest.skip('test requires network connection')
```

Reply:
Great. I hope your improvements can make it back into pytest-socket ;-)